### ISSUE TITLE
Make execute function respect specific dynamic types based on what is passed into the function

### ISSUE THREAD

**Issue by noahjahn** [noahjahn/satisfactory-server-sdk#6]
For example, if I call the `healthcheck` function in the `execute` method of the SF SDK like so:

```typescript
const healthcheck = await satisfactory.execute<
  HealthCheckRequestData,
  HealthCheckResponseBody
>('healthcheck');
```

The two passed in types should be only an allowed set of types based on the fact that I passed in the `'healthcheck'` value.

So, I should _not_ be able to do:

```typescript
const healthcheck = await satisfactory.execute<
  PasswordLoginRequestData,
  PasswordLoginResponseBody
>('healthcheck');
```

**Comment by lundjrl** [comment]
Braindump comment: discriminated unions based on string from enum.

**Comment by noahjahn** [comment]
Implemented in #12
